File: 2020/day02/part2.py
file = [i.split(' ') for i in open("input.txt", "r").read().split("\n")]
res = 0
for i in file:
	num_ran = [int(j) for j in i[0].split('-')]
	if sum(1 for q in num_ran if i[2][q-1] == i[1][0]) == 1:
		res += 1
print(res)

File: 2021/day10/part2.nim
include prelude
import std/algorithm

let
 input = readFile("input.txt").strip.split("\n")
 closepoints = {')': 3, ']': 57, '}': 1197, '>': 25137}.toTable
 openpoints = {'(': 1, '[': 2, '{': 3, '<': 4}.toTable
 openclose = {'(': ')', '[': ']', '{': '}', '<': '>'}.toTable


proc findError(line: string): int =
 var last_open: seq[char]
 for brack in line:
  if brack in openclose:
   last_open.add(brack)
  elif brack == openclose[last_open[^1]]:
   last_open.delete(last_open.high)
  else:
   return -1
 for brack in last_open.reversed:
  result *= 5
  result.inc openpoints[brack]

 

var sum: seq[int]
for line in input:
 let tmp = findError(line)
 if tmp > 0:
  sum.add(tmp)

echo sum.sorted(system.cmp[int])[sum.len div 2]